harness/determined/deploy/gke/cli.py

```diff
@@ -0,0 +1,551 @@
+
+    if s not in valid_accelerator_types:
+        raise argparse.ArgumentTypeError(
+            "Accelerator must be one of {}".format(valid_accelerator_types)
+        )
+
+
+def validate_location(location: str, isZone: bool = True) -> None:
+    try:
+        cmd = ["gcloud", "compute"]
+        if isZone:
+            cmd += ["zones"]
+        else:
+            cmd += ["regions"]
+        cmd += ["describe", location]
+        subprocess.check_call(cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+    except:
+        raise argparse.ArgumentTypeError(
+            "The specified {} {} was not found".format("zone" if isZone else "region", location)
+        )
+
+
+def validate_machine_type(machine_type: str, zone: str) -> None:
+    try:
+        subprocess.check_call(
+            [
+                "gcloud",
+                "compute",
+                "machine-types",
+                "describe",
+                machine_type,
+                "--zone",
+                zone,
+            ],
+            stdout=subprocess.DEVNULL,
+        )
+    except:
+        raise ValueError(
+            "The specified machine type {} was not found in the specified zone {}.".format(
+                machine_type, zone
+            ),
+        )
+
+
+def validate_accelerator_zone(args: argparse.Namespace, zone: str) -> None:
+    try:
+        subprocess.check_call(
+            [
+                "gcloud",
+                "compute",
+                "accelerator-types",
+                "describe",
+                args.gpu_type,
+                "--zone",
+                zone,
+            ],
+            stdout=subprocess.DEVNULL,
+        )
+    except:
+        raise ValueError(
+            "The specified accelerator type {} was not found in the specified zone {}.".format(
+                args.gpu_type, zone
+            ),
+        )
+
+
+def validate_args(args: argparse.Namespace) -> None:
+    validate_location(args.zone, isZone=True)
+    validate_accelerator_type(args.gpu_type)
+    if args.master_machine_type != "n1-standard-16":
+        validate_machine_type(args.master_machine_type, args.zone)
+    if args.agent_machine_type != "n1-standard-32":
+        validate_machine_type(args.agent_machine_type, args.zone)
+    validate_accelerator_zone(args, args.zone)
+    if args.multi_node_pool:
+        if args.cpu_node_pool_name is None:
+            raise ValueError(
+                "--cpu-node-pool-name must be specified if using multiple node pools "
+                + "(--multi-node-pool flag is set).",
+            )
+    if args.gpu_coscheduler and args.preemption:
+        raise ValueError(
+            "--gpu-coscheduler and --preemptive-scheduler are mutually exclusive and cannot both be"
+            " specified",
+        )
+    if args.cpu_coscheduler and not args.gpu_coscheduler:
+        raise ValueError(
+            "To enable coscheduling on the CPU Node Pool, coscheduling must be enabled on the GPU"
```

Comment:
then perhaps just one flag for `--coscheduling` will be sufficient? or are there plans to split these?

Reply:
If using a GPU cluster, I think its possible to use the coscheduler for the GPU but not the CPU, which is why they're separate.

Reply:
There's also a couple of different flavors of coscheduler. There's the one that just provides gang scheduling, there's an even more modified one that has more of a notion of priority, and another one that allows preemption. I'm not sure how many we plan to support long-term or if these are just superceding each other. @eecsliu Can you clarify?